packages/plugin-core/src/views/LookupView.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  assertUnreachable,
+  DendronTreeViewKey,
+  DMessage,
+  LookupModifierStatePayload,
+  LookupViewMessage,
+  LookupViewMessageEnum,
+} from "@dendronhq/common-all";
+import { Logger } from "../logger";
+import * as vscode from "vscode";
+import { WebViewUtils } from "./utils";
+import { LookupControllerV3 } from "../components/lookup/LookupControllerV3";
+import { PickerUtilsV2 } from "../components/lookup/utils";
+import { DendronQuickPickerV2 } from "../components/lookup/types";
+import { DendronBtn, getButtonCategory } from "../components/lookup/buttons";
+import { getExtension } from "../workspace";
+
+export class LookupView implements vscode.WebviewViewProvider {
+  public static readonly viewType = DendronTreeViewKey.LOOKUP_VIEW;
+  private _view?: vscode.WebviewView;
+  private _controller?: LookupControllerV3;
+
+  public postMessage(msg: DMessage) {
+    this._view?.webview.postMessage(msg);
+  }
+
+  public registerController(controller: LookupControllerV3) {
+    this._controller = controller;
+  }
+
+  public deregisterController() {
+    this._controller = undefined;
+  }
+
+  public async resolveWebviewView(
+    webviewView: vscode.WebviewView,
+    _context: vscode.WebviewViewResolveContext<unknown>,
+    _token: vscode.CancellationToken
+  ) {
+    this._view = webviewView;
+
+    WebViewUtils.prepareTreeView({
+      ext: getExtension(),
```

Comment:
[[../packages/plugin-core/src/ExtensionProvider.ts]]

Reply:
I didn't want to introduce noise by adding refactoring of other features (prepareTreeView signature needs to change) in this PR but since it's just one other panel (tree view v2) I'll go ahead and do it here.

Reply:
Looks like the interface `IDendronExtension` is missing a lot of properties that the implementation `DendronExtension` has. (e.g. `DendronExtension.DendronTreeViewV2`, `DendronExtension.treeview`, etc.)

Are these intentionally omitted or is it just in the process of being added?
`ExtensionProvider.getExtension()`'s return type is an `IDendronExtension` so using them results in a type error.

I could ts-ignore them but not ideal, and type casting would drag `DendronExtension` back in so it would defeat the purpose.

If it's just in the process of being added to the interface, I'll go ahead and add the ones that I need here.

Reply:
slightly off-topic, but shouldn't `DendronExtension.DendronTreeViewV2` be part of `DendronExtension.treeview`?

Reply:
> Looks like the interface IDendronExtension is missing a lot of properties that the implementation DendronExtension has. (e.g. DendronExtension.DendronTreeViewV2, DendronExtension.treeview, etc.)

With the refactor, we're taking stock of what properties we want to keep vs prune. If we have missing properties, lets not use `ts-ignore`. can we pass in the full DendronExtension in the constructor inside `[[../packages/plugin-core/src/workspace.ts]]` and use the original interface?

Reply:
> I didn't want to introduce noise by adding refactoring of other features 

Got it. that makes sense. in the future, we would split this off into two commits - a separate one to refactor for extension provider and then rebase this commit off of that one

Reply:
> slightly off-topic, but shouldn't DendronExtension.DendronTreeViewV2 be part of DendronExtension.treeview?

the reason its not is because the web tree view is not accessible and the native one is. have an action item to implement an accessible tree view but in the meanwhile, we're sticking with original tree view

Reply:
> With the refactor, we're taking stock of what properties we want to keep vs prune. If we have missing properties, lets not use ts-ignore. can we pass in the full DendronExtension in the constructor inside [[../packages/plugin-core/src/workspace.ts]] and use the original interface?

Isn't this effectively the same as using the deprecated bare `getExtension()`? We do know that calling the constructor for `LookupView` within the DendronExtension implementation would guarantee that it will be loaded first, but it's not the only place that needs to access properties of the concrete implementation of `DendronExtension`

[lookup controller](https://github.com/dendronhq/dendron/pull/1977/files#diff-c230e5548056948aac47967c667f18558c8764f0b7da3dfee5f9b932be31c30eR127) has to get the `treeview` property of `extension` (not in interface)

[lookup utils](https://github.com/dendronhq/dendron/pull/1977/files#diff-9a073d69e74cc4efb2e01647039fbbe70afe5a539f2e46d39963b85db838c796R728) also needs it for the same reason.

If the decision of keeping treeview related properties in `IDendronExtension` isn't made yet, I think it would make sense to keep it as it was (bare getExtension()) so that this gets cleaned up along with others when we do have a decision in place for where treeview related stuff is going to reside moving forward. No need to change up what is going to be fixed soon.

The two other places I mentioned just above is why I think this PR should wait for #2004 to be in since it seems to be untangling dependencies all across lookup related stuff and here I'm adding more to it.

Reply:
makes sense, lets keep it in place for now